test/tbb/test_concurrent_queue.cpp

```diff
@@ -199,3 +199,96 @@ TEST_CASE("Test exception in allocation") {
 }
 
 #endif // TBB_USE_EXCEPTIONS
+
+class MinimalisticObject {
+public:
+    struct flag {};
+    static constexpr std::size_t default_obj = 42;
+
+    MinimalisticObject() = delete;
+    MinimalisticObject(flag) : underlying_obj(default_obj) {}
+
+    MinimalisticObject(const MinimalisticObject&) = delete;
+    MinimalisticObject& operator=(const MinimalisticObject&) = delete;
+
+    std::size_t get_obj() const { return underlying_obj; }
+
+protected:
+    std::size_t underlying_obj;
+    friend struct MoveAssignableMinimalisticObject;
+};
+
+constexpr std::size_t MinimalisticObject::default_obj;
```

Comment:
static constexpr variables can be initialized inside class, so you don't need move initialization outside class, am I right?

Reply:
If I correctly understood [this](https://stackoverflow.com/questions/8016780/undefined-reference-to-static-constexpr-char) 
we still need to provide definition for static constexpr variable outside of the class until C++17. 
My compiler generates a link-time error if I remove this line. Some compilers don't.
I have reworked this place because it does not strictly required